File: stars-app/frontend/src/types/simulator.ts
import type { Channel, Incentive, GapStatus, PropensityTier } from "./common";

export interface SimulatorConfig {
  contract_id: string;
  measure_code: string | null;
  gap_statuses: GapStatus[];
  propensity_tiers: PropensityTier[];
  channels: Channel[];
  incentive: Incentive;
  suppressions: {
    recently_contacted: boolean;
    opted_out: boolean;
    already_closed: boolean;
  };
}

export interface WaterfallWeek {
  label: string;
  outreach_count: number;
  estimated_closures: number;
  cumulative_compliance: number;
}

export interface SegmentRow {
  label: string;
  members: number;
  channel: string;
  incentive: string;
  estimated_closures: number;
  closure_rate_pct: number;
  estimated_cost: number;
}

export interface SimulatorResult {
  net_pool: number;
  estimated_closures: number;
  projected_rate: number;
  lift: number;
  eligible_members: number;
  measure_name: string;
  measure_pct: number;
  channel_mix: Record<string, number>;
  waterfall: WaterfallWeek[];
  segments: SegmentRow[];
}

File: stars-app/backend/models/simulator.py
from pydantic import BaseModel
from .common import Channel, Incentive, GapStatus


class SimulatorConfig(BaseModel):
    contract_id: str = "H3312"
    measure_code: str | None = None
    gap_statuses: list[GapStatus] = ["Open", "Partial", "Borderline"]
    propensity_tiers: list[str] = ["high", "medium", "low"]
    channels: list[Channel] = ["Call", "SMS", "Email"]
    incentive: Incentive = "None"
    suppressions: dict[str, bool] = {
        "recently_contacted": True,
        "opted_out": True,
        "already_closed": True,
    }


class WaterfallWeek(BaseModel):
    label: str
    outreach_count: int
    estimated_closures: int
    cumulative_compliance: float


class SegmentRow(BaseModel):
    label: str
    members: int
    channel: str
    incentive: str
    estimated_closures: int
    closure_rate_pct: int
    estimated_cost: int


class SimulatorResult(BaseModel):
    net_pool: int
    estimated_closures: int
    projected_rate: float
    lift: float
    eligible_members: int
    measure_name: str
    measure_pct: float
    channel_mix: dict[str, int]
    waterfall: list[WaterfallWeek]
    segments: list[SegmentRow]
